### PR TITLE
Add a dialpad UI for PSTN lookup

### PR DESCRIPTION
**Pull request by dbkr**
Queries the homeserver for PSTN protocol support, and if found,
the add-room button on the DM rooms list section opens a context
menu instead with a 'dial pad' option as well as the current 'start chat'
dialog. Entering a number into this and pressing dial performs
a thirdparty user query for the given string and starts a DM with that
user.

Screenshot 2020-12-23 at 19 05 37
Screenshot 2020-12-23 at 19 05 49

### CODE REVIEW

File: src/components/views/rooms/RoomList.tsx [hunk 1]
```diff
@@ -187,6 +222,10 @@ export default class RoomList extends React.PureComponent<IProps, IState> {
             isNameFiltering: !!RoomListStore.instance.getFirstNameFilterCondition(),
         };
 
+        // shallow-copy from the template as we need to make modifications to it
+        this.tagAesthetics = Object.assign({}, TAG_AESTHETICS);
```
Comment (turt2live) [inline src/components/views/rooms/RoomList.tsx:226]: we have this sort of shallow copy mildly abstracted these days via `objectFastClone` (or similar naming)

File: src/components/views/rooms/RoomList.tsx [hunk 2]
```diff
@@ -202,6 +241,17 @@ export default class RoomList extends React.PureComponent<IProps, IState> {
         if (this.customTagStoreRef) this.customTagStoreRef.remove();
     }
 
+    private updateDmAddRoomAction() {
+        const dmTagAesthetics = Object.assign({}, TAG_AESTHETICS[DefaultTagID.DM]);
```
Comment (turt2live) [inline src/components/views/rooms/RoomList.tsx:245]: `objectFastClone`

File: src/dispatcher/actions.ts [hunk 3]
```diff
@@ -94,4 +94,15 @@ export enum Action {
...
+    /**
+     * Opens the modal dial pad
+     */
+    OpenDialPad = "open_dial_pad",
+
+    /**
+     * Fired when CallHandler has checked for PSTN protocol support
+     * XXX: Is an action the right thing for this?
+     */
+    PstnSupportUpdated = "pstn_support_updated",
```
Comment (turt2live) [inline src/dispatcher/actions.ts:107]: can these also document which payload they expect? (or "No additional payload information required." if none relevant)

**Review by turt2live — APPROVED** [review]
otherwise lgtm

File: src/components/views/rooms/RoomList.tsx (hunk 1 above)
Comment (dbkr) [reply src/components/views/rooms/RoomList.tsx:226]: ah, I'm a bit surprised we prefer custom stuff over native JS, but OK.

**Review by nadonomy — APPROVED** [review]

**Pull request merged**